### PR TITLE
GuildSetting/Caching + EXP

### PR DESCRIPTION
**Pull request by july12123**

### CODE REVIEW

File: src/commands/leveling/Rank.ts [hunk 1]
```diff
@@ -11,17 +11,19 @@ export default class RankCommand extends BaseCommand {
     }
 
     async run(client: DiscordClient, interaction: CommandInteraction) {
-        if (!interaction.guild) return;
-        //remove
-        if(interaction.client){
+        if (!interaction.guildId){
             interaction.followUp({
```
Comment (rohank05) [inline src/commands/leveling/Rank.ts:15]: How about the message: This command can only be used inside server

File: src/database/functions/GuildSettingsFunctions.ts [hunk 2]
```diff
@@ -1,35 +1,56 @@
...
-            expModule: false,
-            newsModule: false,
-            welcomeModule: false,
-            inviteModule: false,
-        });
-    
+        const newGuild = GuildScheme(guild_id)
         newGuild.save().catch(console.error);
-        return newGuild as any;
+        console.log(newGuild)
```
Comment (rohank05) [inline src/database/functions/GuildSettingsFunctions.ts]: remove log

File: src/database/functions/levelingOperation.ts [hunk 3]
```diff
@@ -1,8 +1,55 @@
+import { guildXP,userXP } from "../../utils/interfaces/leveling";
 import { levelXp } from "../schemas";
 
-export async function getLevel(guild?: string) {
-    const guildXP = await levelXp.findOne({ guild });
-    return guildXP as any;
+export async function getLevel(guildID: string) {
+    var guildData = await levelXp.findOne({ guild:guildID })
```
Comment (rohank05) [inline src/database/functions/levelingOperation.ts]: let julio

Comment (norowachi) [reply src/database/functions/levelingOperation.ts]: ~~const noro~~

**Pull request merged**